components/brave_wallet_ui/components/desktop/views/portfolio/components/portfolio-overview-chart/portfolio-overview-chart.tsx

```diff
@@ -25,21 +31,51 @@ export const PortfolioOverviewChart: React.FC<Props> = ({
   portfolioPriceHistory,
   isLoading
 }) => {
+  // redux
+  const hidePortfolioBalances = useSafeWalletSelector(
+    WalletSelectors.hidePortfolioBalances
+  )
+  const defaultFiatCurrency = useSafeWalletSelector(
+    WalletSelectors.defaultFiatCurrency
+  )
+
   // memos
-  const priceHistory = React.useMemo((): TokenPriceHistory[] => {
+  const encodedPriceData = React.useMemo(() => {
     if (hasZeroBalance || !portfolioPriceHistory) {
-      return []
+      return encodeURIComponent(
+        JSON.stringify({
+          defaultFiatCurrency,
+          hidePortfolioBalances,
+          priceData: []
+        })
+      )
     }
-    return portfolioPriceHistory
-  }, [portfolioPriceHistory, hasZeroBalance])
+
+    const iframeData: LineChartIframeData = {
+      defaultFiatCurrency,
+      hidePortfolioBalances,
+      priceData: portfolioPriceHistory
+    }
+
+    return encodeURIComponent(JSON.stringify(iframeData))
+  }, [
+    portfolioPriceHistory,
+    hasZeroBalance,
+    defaultFiatCurrency,
+    hidePortfolioBalances
+  ])
 
   // render
   return (
     <Column alignItems='center' fullWidth>
-      <LineChart
-        priceData={priceHistory}
-        isLoading={hasZeroBalance ? false : isLoading}
-        isDisabled={hasZeroBalance}
+      <iframe
+        width={'100%'}
+        height={'130px'}
+        frameBorder={0}
+        src={`chrome-untrusted://line-chart-display${
+          isLoading ? '' : `?${encodedPriceData}`
+        }`}
+        sandbox='allow-scripts allow-same-origin'
```

Comment:
not really an issue but just curious why allow-same-origin is needed?

Reply:
Not needed. Removed. Thanks for catching this